packages/extension/src/background/index.ts

```diff
@@ -15,6 +27,95 @@ const cacheAmplitudeDeviceId = async ({
   }
 };
 
+const excludedCompanionOrigins = [
+  'https://twitter.com',
+  'https://www.google.com',
+  'chrome-extension://',
+];
+
+const isExcluded = (origin: string) =>
+  excludedCompanionOrigins.some((e) => origin.includes(e));
+
+const sendBootData = async (req, sender) => {
+  if (isExcluded(sender?.origin)) {
+    return;
+  }
+  const cacheData = JSON.parse(localStorage.getItem('boot:local'));
+  const { data, settings } = await getBootData('companion', sender?.tab?.url);
+  let settingsOutput = settings;
+  if (!cacheData?.user || !('providers' in cacheData?.user)) {
+    settingsOutput = { ...settingsOutput, ...cacheData?.settings };
+  }
+  await browser.tabs.sendMessage(sender?.tab?.id, {
+    data,
+    settings: settingsOutput,
+  });
+};
+
+function handleMessages(message, sender) {
+  if (message.type === 'CONTENT_LOADED') {
+    sendBootData(message, sender);
+    return null;
+  }
+
+  if (message.type === 'UPVOTE_POST') {
```

Comment:
A reminder that we want to change the API with the background worker to be something like a proxy to the `request` function rather than implementing a new type for every graphql request.